modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+        Assert.assertNull(scimResponse.getHeaderParamMap());
+    }
+
+    @DataProvider(name = "dataForTestDeleteUserFails")
+    public Object[][] dataToTestDeleteUserFails()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        return new Object[][]{
+                {id, ResponseCodeConstants.CODE_INTERNAL_ERROR},
+                {id, ResponseCodeConstants.CODE_BAD_REQUEST},
+                {id, ResponseCodeConstants.CODE_RESOURCE_NOT_FOUND},
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteUserFails")
+    public void testDeleteUserFails(String id, int expectedScimResponseStatus)
+            throws NotFoundException, NotImplementedException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/" + USER_ID);
+
+        if (expectedScimResponseStatus == ResponseCodeConstants.CODE_INTERNAL_ERROR) {
+
+            abstractResourceManager.when(()
+                    -> AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+            SCIMResponse scimResponse = userResourceManager.delete(id, null);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else if (expectedScimResponseStatus == ResponseCodeConstants.CODE_BAD_REQUEST) {
+
+            abstractResourceManager.when(()
+                    -> AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+            doThrow(new BadRequestException()).when(userManager).deleteUser(id);
+            SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        } else {
+
+            abstractResourceManager.when(()
+                    -> AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+            doThrow(new NotFoundException()).when(userManager).deleteUser(id);
+            SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+            Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        }
+    }
+
+    @DataProvider(name = "dataForTestDeleteUserFailsINCharonExceptionOnly")
+    public Object[][] dataToTestDeleteUserCharonExceptionOnly()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String id = user.getId();
```

Comment:
```suggestion
        String id = getNewUser().getId();
```

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11